lib/exchange/currencies.rb

```diff
@@ -119,67 +187,137 @@ def stringify amount, currency, opts={}
     # iso definition
     # @param [BigDecimal, Fixed, Float, String] amount The amount of money you want to round
     # @param [String, Symbol] currency The currency you want to round the money in
     # @example Round a currency with 2 minors
-    #   Exchange::ISO.round("4523.456", "usd") #=> #<Bigdecimal 4523.46>
-    
+    #   Exchange::Currencies.round("4523.456", "usd") #=> #<Bigdecimal 4523.46>
+
     install_operation :round
-    
-    # Use this to ceil a currency amount. This allows us to ceil exactly to the number of minors the currency has in the 
+
+    # Use this to ceil a currency amount. This allows us to ceil exactly to the number of minors the currency has in the
     # iso definition
     # @param [BigDecimal, Fixed, Float, String] amount The amount of money you want to ceil
     # @param [String, Symbol] currency The currency you want to ceil the money in
     # @example Ceil a currency with 2 minors
-    #   Exchange::ISO.ceil("4523.456", "usd") #=> #<Bigdecimal 4523.46>
-    
+    #   Exchange::Currencies.ceil("4523.456", "usd") #=> #<Bigdecimal 4523.46>
+
     install_operation :ceil
-    
-    # Use this to floor a currency amount. This allows us to floor exactly to the number of minors the currency has in the 
+
+    # Use this to floor a currency amount. This allows us to floor exactly to the number of minors the currency has in the
     # iso definition
     # @param [BigDecimal, Fixed, Float, String] amount The amount of money you want to floor
     # @param [String, Symbol] currency The currency you want to floor the money in
     # @example Floor a currency with 2 minors
-    #   Exchange::ISO.floor("4523.456", "usd") #=> #<Bigdecimal 4523.46>
-    
+    #   Exchange::Currencies.floor("4523.456", "usd") #=> #<Bigdecimal 4523.46>
+
     install_operation :floor
-    
+
     # Forwards the assure_time method to the instance using singleforwardable
     #
-    def_delegators :instance, :definitions, :instantiate, :stringify, :symbol, :round, :ceil, :floor, :currencies, :country_map, :defines?, :assert_currency!
-    
+    def_delegators :instance, :definitions, :instantiate, :stringify, :symbol,
+      :round, :ceil, :floor, :currencies, :country_map, :defines?,
+      :assert_currency!, :iso4217_definitions, :crypto_definitions,
+      :historical_definitions, :active_definitions, :oxr_definitions,
+      :iso4217_currencies, :crypto_currencies, :historical_currencies,
+      :active_currencies, :oxr_currencies
+
     private
-    
+
+    def tag_all(hsh, tag)
+      hsh.each do |k, v|
+        hsh[k][:tags] = [] if v[:tags].nil?
+        hsh[k][:tags] += [tag]
+      end
+
+      hsh
+    end
+
+    # Load the currencies from their file, with a quick sanity check that
+    # any `replaced_by` references are included in the file
+    def load_currencies
+      iso_currencies = symbolize_keys(
+        YAML.load_file(File.join(ROOT_PATH, 'iso4217.yml'))
+      )
+      tag_all(iso_currencies, :iso4217)
```

Comment:
Do we need this notion of tagging? Can we not just load the files into their respective collections, and concat collections as needed?

Reply:
Yeah, can do.